console/src/state/resources.rs

```diff
@@ -158,22 +166,53 @@ impl ResourcesState {
             };
 
             let id = resource.id?.id;
+            let parent_id = resource.parent_resource_id.map(|id| id.id);
+
+            let parent = strings.string(match parent_id {
+                Some(id) => parents
+                    .get(&id)
+                    .and_then(|r| r.upgrade())
+                    .map(|r| {
+                        let r = r.borrow();
+                        format!("{} ({}::{})", r.id(), r.target(), r.concrete_type())
+                    })
+                    .unwrap_or(format!("{}", id)),
```

Comment:
nit: this should probably be 
```suggestion
                    .unwrap_or_else(|| format!("{}", id)),
```
so that we don't format the string if we _did_ find a parent (see the documentation for https://doc.rust-lang.org/stable/core/option/enum.Option.html#method.unwrap_or)

Reply:
also, it might be _slightly_ more efficient not to go through the `format!` macro when the format arguments are just `"{}"`; we can use the `fmt::Display` implementation for `u64` directly:
```suggestion
                    .unwrap_or_else(|| id.to_string()),
```

but, that's probably not a _huge_ performance impact, regardless.